regression/synthetic_dataset.py

```diff
@@ -1,200 +1,306 @@
 
-def delete_oci(remote_dir: str) -> None:
-    """Delete a remote directory from oci.
+    args, runtime_args = args.parse_known_args()
+    kwargs = {get_kwargs(k): v for k, v in zip(runtime_args[::2], runtime_args[1::2])}
+    return args, kwargs
 
-    Args:
-        remote_dir (str): Location of the remote directory.
-    """
-    import oci
 
-    obj = urllib.parse.urlparse(remote_dir)
+def get_dataset_params(kwargs: Dict[str, str]) -> Dict[str, Any]:
+    """Get the dataset parameters from command-line arguments.
 
-    config = oci.config.from_file()
-    oci_client = oci.object_storage.ObjectStorageClient(
-        config=config, retry_strategy=oci.retry.DEFAULT_RETRY_STRATEGY)
-    namespace = oci_client.get_namespace().data
-    objects = oci_client.list_objects(namespace, obj.netloc, prefix=obj.path.lstrip('/'))
+    Args:
+        kwargs (Dict[str, str]): Command-line arguments.
 
-    for filenames in objects.data.objects:
-        oci_client.delete_object(namespace, obj.netloc, filenames.name)
+    Returns:
+        Dict[str, Any]: Dataset parameters.
+    """
+    dataset_params = {}
+    if 'num_samples' in kwargs:
+        dataset_params['num_samples'] = int(kwargs['num_samples'])
+    if 'seed' in kwargs:
+        dataset_params['seed'] = int(kwargs['seed'])
+    if 'shape' in kwargs:
+        dataset_params['shape'] = ast.literal_eval(kwargs['shape'])
+    if 'column_names' in kwargs:
+        dataset_params['column_names'] = ast.literal_eval(kwargs['column_names'])
+    return dataset_params
 
 
-def main(args: Namespace) -> None:
-    """Benchmark time taken to generate the epoch for a given dataset.
+def main(args: Namespace, kwargs: Dict[str, str]) -> None:
+    """Create and delete a dataset.
 
     Args:
-        args (Namespace): Command-line arguments.
+        args (Namespace): Arguments.
+        kwargs (Dict[str, str]): Named arguments.
     """
-    remote_dir = args.cloud_url if args.cloud_url is not None else utils.get_local_remote_dir()
+    dataset_params = get_dataset_params(kwargs)
+    writer_params = get_writer_params(kwargs)
+    if args.name.lower() not in DATASET_MAP:
+        raise ValueError(f'Unsupported dataset {args.name}. Supported: {DATASET_MAP.keys()}')
+    dataset = DATASET_MAP[args.name.lower()](**dataset_params)
+    columns = {name: dtype for name, dtype in zip(dataset.column_names, dataset.column_encodings)}
+
     if args.create:
-        dataset = get_dataset(args.num_samples)
-        with MDSWriter(
-                out=remote_dir,
-                columns=_COLUMNS,
-                compression=args.compression,
-                hashes=args.hashes,
-                size_limit=args.size_limit,
-        ) as out:
+        with MDSWriter(out=args.out, columns=columns, **writer_params) as out:
             for sample in dataset:
                 out.write(sample)
-    if args.delete:
-        obj = urllib.parse.urlparse(remote_dir)
+    elif args.delete:
```

Comment:
If this is elif, probably want to specify that only --create or --delete can be used and add an assert for that

Reply:
Done by using `ArgumentParser.add_mutually_exclusive_group()`